### PR TITLE
Add Double Cubic Lattice Volume (DCLV).

### PR DESCRIPTION
**Pull request by RPirie96**
Method to compute the solvent accessible surface area and volume, van der Waals volume, compactness and packing density of a protein. Implemented from:  Frank Eisenhaber, Philip Lijnzaad, Patrick Argos, Chris Sander and Michael Scharf, "The Double Cubic Lattice Method: Efficient Approaches to Numerical Integration of Surface Area and Volume and to Dot Surface Contouring of Molecular Assemblies", Journal of Computational Chemistry, Vol. 16, No. 3, pp. 273-284, 1995.

#### What does this implement/fix? Explain your changes.
New method based on a chunk of old code from Roger for calculating surface area and volume from a dot surface. Primarily for proteins, but hopefully a useful addition :)

### CODE REVIEW

**Comment by greglandrum** [comment]
Hi @RPirie96 I haven't looked at the code yet, but it certainly sounds interesting.
I'm curious why you decided to put it in `External` instead of `GraphMol`.
We normalize use `External` for interfacing with code from another project. In this case the library is self-contained, so I think it could be either directly under `GraphMol` or, perhaps, part of `GraphMol/Descriptors`

**Comment by RPirie96** [comment]
Hi @greglandrum  sorry that was just me not knowing where to put it! I've moved it across to GraphMol/Descriptors, and the tests now all pass so should be good for review

**Comment by greglandrum** [comment]
@RPirie96 I just did a PR against your branch with a bunch of suggested changes.

File: Code/GraphMol/Descriptors/DCLV.h [hunk 1]
```diff
@@ -0,0 +1,86 @@
+#ifndef RDKIT_DCLV_H
```
Comment (greglandrum) [inline Code/GraphMol/Descriptors/DCLV.h:1]: I forgot this in my other suggestions, but please add a copyright statement modelled on other RDKit files to this and DCLV.cpp

**Review by greglandrum — CHANGES_REQUESTED** [review]
@RPirie96 : the only thing this is missing is the copyright statements in DCLV.h and DCLV.cpp

If you can get those added in the next few days, then we can add this to the upcoming 2024.03.1 release, which would be cool. :-)

**Comment by RPirie96** [comment]
Hi @greglandrum apologies, I missed the original comment about copyright statements - added now so should be good to go.

**Review by greglandrum — APPROVED** [review]
LGTM

**Comment by greglandrum** [comment]
Thanks for this contribution @RPirie96. 
I will merge it once the CI builds go through.

**Pull request merged**